title: \sf Lecture 1 - RNA-Seq analysis step-by-step
header-includes:
- \usepackage{color}
- \usepackage{float}
- \DeclareUnicodeCharacter{2212}{-}
date: "Sept. 6, 2023"
output:
  bookdown::html_document2:
    toc: true
fontsize: 14pt
geometry: margin=1in

# Introduction to RNA-seq
RNA-Seq data of the short read variety is typically generated on Illumina machines. The raw data for RNA-Seq datasets from "all" publications on NCBI are deposited online on sites such as Gene Expression Omnibus (GEO) [https://www.ncbi.nlm.nih.gov/geo/](https://www.ncbi.nlm.nih.gov/geo/) 

In this tutorial we will download raw .fastq files for an RNA-seq dataset from GEO, generate .bam files for visualization, and perform downstream analysis for differential expression.

## Obtaining FASTQ files using XX

# TODAY: de novo Motif searching with MEME

## MEME: Multiple Em for Motif Elucidation


MEME Website: [http://meme-suite.org/](http://meme-suite.org/)

knitr::include_graphics("./slop.png") 

```{r engine='bash', eval=F, echo=TRUE}
fastaFromBed -fi GENOME.fa -bed SUMMITS_100bp.bed -fo SUMMITS_100bp.fa
#-name: Will use the 'name' column from bed file to name sequence
#-fi: Input fasta sequences to retrieve interval sequences from
#-bed: Input bed file
#-fo: Output fasta file
```

